dev/InfoBar/InfoBarPanel.cpp

```diff
@@ -19,20 +19,28 @@ winrt::Size InfoBarPanel::MeasureOverride(winrt::Size const& availableSize)
     const auto parent = this->Parent().try_as<winrt::FrameworkElement>();
     const float minHeight = !parent ? 0.0f : (float)(parent.MinHeight() - (Margin().Top + Margin().Bottom));
 
-    for (winrt::UIElement const& child : Children())
+    const auto children = Children();
+    const auto childCount = (int)children.Size();
+    for (winrt::UIElement const& child : children)
     {
         child.Measure(availableSize);
         const auto childDesiredSize = child.DesiredSize();
 
         if (childDesiredSize.Width != 0 && childDesiredSize.Height != 0)
         {
             // Add up the width of all items if they were laid out horizontally
-            const auto horizontalMargin = winrt::InfoBarPanel::GetSpacingInHorizontalOrientation(child);
-            totalWidth += childDesiredSize.Width + (nItems > 0 ? (float)horizontalMargin.Left : 0) + (float)horizontalMargin.Right;
+            const auto horizontalMargin = winrt::InfoBarPanel::GetHorizontalOrientationMargin(child);
+            // Ignore left margin of first and right margin of last child
+            totalWidth += childDesiredSize.Width +
```

Comment:
Isn't this supposed to ignore the first/last margins of the items that  aren't collapsed? So we should do a check for non-zero childDesiredSize.Width?

Reply:
The if statement above this already removes them:
if (childDesiredSize.Width != 0 && childDesiredSize.Height != 0)